nautobot/docs/user-guides/git-data-source.md

```diff
@@ -0,0 +1,257 @@
+
+From the default Git repositories view:
+![](./images/git-data-source_8.png)
+
+From the detailed view:
+![](./images/git-data-source_9.png)
+
+> Once the repository has been synced its easy to check the history for the templates.
+Navigate to **Git Repositories** and select the repository in question.  Once you're in the detailed view you can look through the **Synchronization Status** or **Change Log** tabs.
+
+Now that the Git repository is linked for export templates it can be controlled via the normal Git operations workflow, which allows users or groups of users to perform code reviews using Pull Requests etc.
+
+### Jobs
+Jobs are a way for users to execute custom logic on demand from within the Nautobot UI. Jobs can interact directly with Nautobot data to accomplish various data creation, modification, and validation tasks.
+
+For technical details on jobs, please see the documentation on [jobs](https://nautobot.readthedocs.io/en/latest/additional-features/jobs/#jobs).
+
+Jobs allow a user to write scripts in Python.  By integrating the scripts with Git, a user can utilize Git workflows to manage source control, versioning, and pipelines.
+
+Setting up the repository can be done following the same steps from [Export Templates](#Export-Templates).  The only differences is the `provides` selection changes to `jobs`.
+
+Jobs need to be defined in `/jobs/` directory at the root of a Git repository.
+
+An example tree for `/jobs/`.
+
+```
+▶ tree jobs 
+jobs
+├── __init__.py
+└── get-device-connection.py
+
+1 directory, 2 files
+```
+
+Once the repository is created in Nautobot.
+![](./images/git-data-source_10.png)
+
+> Note: The same repository and branch can be used for the different `provides` methods.  Nautobot Git as a data source will look for specific root directory names.
+
+Once the scripts have been pushed into the repository, a sync needs to be executed, after which navigating to Jobs via **Extensibility -> Jobs** will show the new jobs loaded from the Git repository.
+
+![](./images/git-data-source_11.png)
+
+Jobs now shows the job from the Git repository.
+
+![](./images/git-data-source_12.png)
+
+At this point all changes, and history can be kept using Git.  A simple `sync` operation can be done from Nautobot to pulldown any changes.
+
+### Config Contexts
+Detailed information on [config contexts](https://nautobot.readthedocs.io/en/latest/models/extras/gitrepository/#configuration-contexts).
+
+Config contexts may be provided as JSON or YAML files located in `/config_contexts/`.
+
+Config contexts can be used to provide additional details to different automation tooling.  For example `Ansible vars`, or any other data that you can't natively store in Nautobot.  It can also be used in the Golden Configuration Nautobot plugin to provide extra details to generate configuration templates.
```

Comment:
```suggestion
Config contexts can be used to provide additional details to different automation tooling.  For example Ansible variables, or any other data that you can't natively store in Nautobot.  It can also be used in the Golden Configuration Nautobot plugin to provide extra details to generate configuration templates.
```